# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.h
# Commit: 2fe2afdee09d
# Full Hash: 2fe2afdee09dd5d18d7208f5b4186e659ebdde7e
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-02 21:26:33
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 11: Don't preload anchored links from early hints r=necko-reviewers,valentin
#   
#   They won't get passed to the PreloadService anyway. Resulting in an unnecessary
#   additional load
#   
# ==============================================================================

diff -r 8bdb1f682d22 -r 2fe2afdee09d dom/base/nsContentUtils.h
--- a/dom/base/nsContentUtils.h	Fri Dec 02 09:45:27 2022 +0000
+++ b/dom/base/nsContentUtils.h	Fri Dec 02 09:45:28 2022 +0000
@@ -1409,6 +1409,12 @@
       nsContentPolicyType aType);
 
   /**
+   * check whether the Link header field applies to the context resource
+   * see <http://tools.ietf.org/html/rfc5988#section-5.2>
+   */
+  static bool LinkContextIsURI(const nsAString& aAnchor, nsIURI* aDocURI);
+
+  /**
    * Returns true if the content policy type is any of:
    *   * TYPE_INTERNAL_SCRIPT_PRELOAD
    *   * TYPE_INTERNAL_IMAGE_PRELOAD